### ISSUE TITLE
tf frames jumping around for chest_camera

### ISSUE THREAD

**Issue by ToMadoRe** [strands-project/strands_movebase#6]
@nilsbore: I calibrated the chest_camera as described in the README. When I visualize the chest cam's pointcloud of the floor in RVIZ, the plane jumps around like hell (fixed frame: base_link).

**Comment by nilsbore** [comment]
Are the tf frames jumping around or is the actual point cloud jumping around? The tf frames have always been jumping in visualization as far as I know. But if there's a new problem it could be related to the decoupling of the tf state publisher.

**Comment by nilsbore** [comment]
Is it jumping distances of meters or just a lag in the update when the robot is moving?

**Comment by nilsbore** [comment]
Oh, I probably know why this is happening. You're still running the old scitos_common but the new strands_movebase. Please update scitos_common!

**Comment by ToMadoRe** [comment]
both is jumping around. The point cloud of the floor plane is once vertically and a second later horizontally, then something inbetween,... I put a CATKIN_IGNORE in strands_movebase (because of the duplicated executables) so I am actually using the old scitos_2d_navigation stuff (with the new strands_description package).

As we have a demo tmrw and nothing really works, I am using an old catkin_workspace for now.

**Comment by nilsbore** [comment]
So: are you using the old scitos_common or the new one?

**Comment by nilsbore** [comment]
I.e. the one that was merged yesterday, the previous one also published the chest camera transforms, which is done in strands_description now. If they are both run together, it would probably give those effects.

**Comment by ToMadoRe** [comment]
I am not at the robot any more - not really sure if updated scitos_common as well. I tried quite a lot of things to get our system running... But at one point, I had the problem that there was no transform for the chest_link being published. I guess that would mean that scitos_common was at its latest state, right?

**Comment by nilsbore** [comment]
Yes, that would probably mean that you had the new package. I will try to replicate this on Rosie tomorrow. Can you try and see which `scitos_state_publisher.launch` is launched when these effects happen.

**Comment by ToMadoRe** [comment]
I launched strands_robot in scitos_bringup (i.e. including scitos_state_publisher and using the released scitos_mira package as well as the "new" sicks300 driver) and in addition launched the strands_state_publisher to get the chest links. I also started map_saver because I wanted to create a new map. But RVIZ just showed a jumping plane and complained about missing /map frame transforms when trying to visualize the map in RVIZ. At this point, I gave up and so I am not really sure what's causing the problem.

If I can get our robot running with tjhe backup  workspace, then I will probably not change anything for the next two days. But I will keep you posted if I find out more.

**Comment by nilsbore** [comment]
Ok, do you know which scitos_state_publisher is launched? Were you running gmapping?

I'll try to get to the bottom of this, didn't have these problems when I tried the new version on the robot.

**Comment by nilsbore** [comment]
I.e. are you getting the chest_camera frames before you launch strands_state_publisher and are the tf frames shaking before that.

**Comment by nilsbore** [comment]
I can't replicate this on Rosie so I'm still suspecting that it's somehow launching the old scitos_state_publisher on Werner.

**Comment by ToMadoRe** [comment]
I just checked scitos_common again and it is up to date. I have a some CATKIN_IGNOREs though: in 
scitos_msgs
`scitos_common/scitos_common`
`strands_movebase/strands_movebase`

And I launched  `scitos_description scitos_state_publisher.launch` and `strands_description strands_state_publisher.launch`

But let us check that next week. I think I got the marathon system running with an old strands workspace and don't want to change anything anymore.
